SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +658,265 @@ private void DisplayErrorMessage(string message)
+
+            if (CheckRegisterData(password, username))
+            {
+                string salt = CreateSalt(saltSize);
+                csvData.AddData(new UserData(username, GenerateHash(password, salt), salt, GenerateUserID()));
+                csvData.SaveData();
+
+                UsernameTextBoxR.Text = "";
+                PasswordBoxR.Password = "";
+                ConfirmPasswordBoxR.Password = "";
+
+                User.Visibility = Visibility.Collapsed;
+                LoginWhenRegistered.Visibility = Visibility.Visible;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void LoginWhenRegistered_Clicked(object sender, RoutedEventArgs e)
+        {
+            GridLogin.Visibility = Visibility.Visible;
+            GridRegistration.Visibility = Visibility.Collapsed;
+        }
+
+        private int GenerateUserID()
+        {
+            Random rnd = new Random();
+            int randomID = rnd.Next(1000, 9999);
+
+            if (Duplicates(randomID))
+            {
+                return GenerateUserID();
+            }
+            return randomID;
+        }
+
+        public bool Duplicates(int randomID)
+        {
+            foreach (var userID in csvData.GetAllData())
+            {
+                if (randomID.Equals(userID.ID))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public bool CheckRegisterData(string password, string username)
+        {
+            if(password.Length == 0 || username.Length == 0 || ConfirmPasswordBoxR.Password.Length == 0)
+            {
+                LabelForEmptyFieldsError.Visibility = Visibility.Visible;
+                return false;
+            }
+            else if (csvData.GetData(UsernameTextBox.Text).Username != "")
+            {
+                LabelForUsedUsernameError.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                if(!CheckUsernameRegex(username))
+                {
+                    LabelForInvalidUsernameError.Visibility = Visibility.Visible;
+                }
+                else // if username is correct
+                {
+                    if(!CheckPasswordLength(password))
+                    {
+                        LabelForPasswordError.Visibility = Visibility.Visible;
+                    }
+                    else if(CheckConfirmPassword(PasswordBoxR.Password, ConfirmPasswordBoxR.Password)) // if password is correct
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private bool CheckPasswordLength(String password)
+        {
+            if (password.Length >= 6)
+            {
+                return true;
+            }
+            return false;
+        }
```

Comment:
This can be just 
private bool CheckPasswordLength(String password) => return password.Length >= 6;